domain/wollemi/service_format.go

```diff
@@ -19,74 +19,136 @@ const (
 	BUILD_FILE = "BUILD.plz"
 )
 
-func (this *Service) Format(config wollemi.Config, paths []string) error {
-	config.Gofmt.Rewrite = wollemi.Bool(false)
+func newGoFormat(paths []string) *goFormat {
+	return &goFormat{
+		resolveLimiter: NewChanFunc(1, 0),
+		isGoroot:       map[string]bool{},
+		paths:          paths,
+		directories:    map[string]*Directory{},
+		external:       map[string][]string{},
+		internal:       map[string]string{},
+		genfiles:       map[string]string{},
+	}
+}
 
-	return this.GoFormat(config, paths)
+// goFormat contains all the state for formatting go rules i.e. the import path to target mapping, and the rules in the
+// targeted Please packages.
+type goFormat struct {
+	// isGoroot contains a cache of import paths that are part of the Go SDK
+	isGoroot map[string]bool
+
+	// paths are the normalised paths we were asked to format
+	paths []string
+
+	// resolveLimiter is used to control the concurrency on resolving targets.
+	resolveLimiter *ChanFunc
+
+	// directories represent the set of Please packages we have parsed
+	directories map[string]*Directory
+
+	// external is a map of third party imports to build targets
+	external map[string][]string
+
+	// internal is a map of this projects imports paths to targets
+	internal map[string]string
+
+	// genfiles contain a map of generated files added via go_copy() to their build targets
+	genfiles map[string]string
```

Comment:
I didn't really understand what this is about. I guess this is something internal to your codebase?